tests/test_spark_launcher.py

```diff
@@ -5,6 +5,15 @@
 __author__ = 'mniekerk'
 
 
+def test_sparkconf_hasattr():
+    c = sparklauncher.SparkConfiguration()
+    assert hasattr(c, "foo") is False
+    assert hasattr(c, "driver_memory") is True
+    assert hasattr(c, "driver-memory") is True
+    assert c.driver_memory is None
+    c.driver_memory = "4g"
+    assert c.driver_memory == "4g"
```

Comment:
I think the set above is reasonable. Maybe check that it raises exception when you try to get a property that is unknown? Maybe try to set a property that is unknown and then try to get it?